File: python/problem2.py

Fn_1 = 0;
Fn_2 = 1;
Fn = 0;
totalSum = 0;
while (Fn < 4000000):
	Fn = Fn_1 + Fn_2
	if (Fn%2 == 0):
		totalSum = totalSum + Fn;
	Fn_1 = Fn_2;
	Fn_2 = Fn;
	
print(totalSum);
	
	

File: javascript/problem2.js
var 
sys = require("sys"),

even = function(n) {
	return (n%2) ? false : true;
},

MAX_FIB = 4000000,

fn_1 = 0, fn_2 = 1, fn = 0, sum = 0; 

while (fn < MAX_FIB) {
	fn = fn_1 + fn_2;
	if (even(fn)){
		sum = sum + fn;
	}
	fn_1 = fn_2;
	fn_2 = fn;
}

sys.puts(sum);
